WordPress/Classes/Services/NotificationSyncService.swift

```diff
@@ -0,0 +1,304 @@
+import Foundation
+
+
+// MARK: - Notifications
+//
+let NotificationSyncServiceDidUpdateNotifications = "NotificationSyncServiceDidUpdateNotifications"
+
+
+// MARK: - NotificationSyncService
+//
+class NotificationSyncService
+{
+    /// Returns the Main Managed Context
+    ///
+    private var contextManager = ContextManager.sharedInstance()
```

Comment:
I've initially attempted doing so (let + reference to the mainMOC), but i'm afraid we need some methods provided by `ContextManager` (such as `newDerivedContext`), and by wiring thing up this way, open the door for dependency injection.

That aside, problem by switching to `let` is: you can't bail early in the initializer, if a certain condition is not met (if you've got *let* properties). Absolute downer!!